werkzeug/_reloader.py

```diff
@@ -197,7 +204,9 @@ def run(self):
                     try:
                         watches[path] = observer.schedule(
                             self.event_handler, path, recursive=True)
-                    except OSError:
+                    except OSError as e:
+                        if e.message == "inotify watch limit reached":
```

Comment:
I think we should special case "path is not a directory" and log any other error messages.